File: 694-Collatz/collatz.cpp
#include <cstdio>
#include <climits>

int numTerms(int A, int L);

int main(){
	int A,L, c = 1;
	while (scanf("%d %d",&A,&L),A >= 0 && L >= 0){
		if (c > 1) printf("\n");
		printf("Case %d: A = %d, limit = %d, number of terms = %d", c++, A, L, numTerms(A,L));
	}
	return 0;
}

int numTerms(int A, int L){
	int count = 1;
	int current = A;
	while (current != 1){
		if (current % 2){
			if (current >= INT_MAX/3)
				break;	
			current = 3*current + 1;
		}
		else
			current /= 2;
		if (current <= L)
			count++;
		else
			break;
	}
	return count;
}

File: UVA/694-Collatz/collatz.py
import sys

def collatz_terms(A, limit):
    num_terms = 0
    while A != 1 and A <= limit:
        if A % 2 == 0:
            A /= 2
        else:
            A = 3 * A + 1
        num_terms += 1
    if A == 1:
        num_terms += 1
    return num_terms

case = 1
for line in sys.stdin:
    A, limit = [int(n) for n in line.split()]
    if A < 0 and limit < 0:
        break
    print("Case {}: A = {}, limit = {}, number of terms = "
          "{}".format(case,A,limit,collatz_terms(A,limit)))
    case += 1

